test/fit_tests/deploy/rackhd_stack_init.py

```diff
@@ -44,22 +55,31 @@ def test01_preload_sku_packs(self):
         for subdir, dirs, files in os.walk('on-skupack/tarballs'):
             for skupacks in files:
                 print "\n**** Loading SKU Pack for " + skupacks
-                fit_common.rackhdapi("/api/1.1/skus/pack", action="binary-post",
+                fit_common.rackhdapi("/api/2.0/skus/pack", action="binary-post",
                                      payload=file(fit_common.TEST_PATH + "on-skupack/tarballs/" + skupacks).read())
             break
         print "\n"
         # check SKU directory against source files
-        errorcount = 0
-        skulist = fit_common.json.dumps(fit_common.rackhdapi("/api/2.0/skus")['json'])
+        errorcount = ""
+        skulist = json.dumps(fit_common.rackhdapi("/api/2.0/skus")['json'])
         for subdir, dirs, files in os.walk('on-skupack'):
             for skus in dirs:
-                if skus not in ["debianstatic", ".git", "packagebuild", "tarballs"] and os.path.isfile('on-skupack/' + skus + '/config.json'):
-                    configfile = fit_common.json.loads(open("on-skupack/" + skus  + "/config.json").read())
-                    if configfile['name'] not in skulist:
-                        print "FAILURE - Missing SKU: " + configfile['name']
-                        errorcount += 1
+                if skus not in ["debianstatic", ".git", "packagebuild", "tarballs"] and \
+                   os.path.isfile('on-skupack/' + skus + '/config.json'):
+                    try:
+                        json.loads(open("on-skupack/" + skus  + "/config.json").read())
+                    except:
+                        # Check is the sku pack config.json file is valid format, fails skupack install if invalid
+                        print "FAILURE - Corrupt config.json in SKU Pack: " + str(skus) + " - not loaded"
+                        errorcount += "  Corrupt config.json in SKU Pack: " + str(skus)
+                    else:
```

Comment:
Updated the code for this as suggested.  Tested and here is a snippet of how output will look if we hit either of the errors:
```
**** Loading SKU Pack for cisco-nexus-C3132_cisco-nexus-C3132.tar.gz
**** Loading SKU Pack for sinetica-hawk-i2_sinetica-hawk-i2.tar.gz
**** Loading SKU Pack for cisco-nexus-C9372PX-E_cisco-nexus-C9372PX-E.tar.gz
**** Loading SKU Pack for brocade-VDX-6740T_brocade-VDX-6740T.tar.gz

FAILURE - Corrupt config.json in SKU Pack: brocade-VDX-6740 - not loaded
FAILURE - Missing SKU: Cisco Nexus 9372PX-E
FAILURE - Missing SKU: Quanta D51 2U
FAILURE - Missing SKU: Quanta D51 1U
FAILURE - Corrupt config.json in SKU Pack: brocade-VDX-6740T - not loaded
F
======================================================================
FAIL: test01_preload_sku_packs (rackhd_stack_init_post.rackhd_stack_init)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/emc/hohene/sandbox-github/rackhd/eh_pollers/test/fit_tests/deploy/rackhd_stack_init_post.py", line 80, in test01_preload_sku_packs
    self.assertEqual(errorcount, "", errorcount)
AssertionError:   Corrupt config.json in SKU Pack: brocade-VDX-6740  Missing SKU: Cisco Nexus 9372PX-E  Missing SKU: Quanta D51 2U  Missing SKU: Quanta D51 1U  Corrupt c\
onfig.json in SKU Pack: brocade-VDX-6740T
```